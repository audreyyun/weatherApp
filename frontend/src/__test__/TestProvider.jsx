import React from 'react'
import { Provider } from 'react-redux';
import configureStore from '../store/store';

const store = configureStore();

const TestProvider = ({ children }) => (
    <Provider store={store}>
        {children}
    </Provider>
);

export default TestProvider;