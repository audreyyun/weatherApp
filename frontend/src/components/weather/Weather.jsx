import React from 'react';
import LocationForm from '../locationForm/LocationForm'
import WeatherInfo from '../WeatherInfo/WeatherInfo'

const Weather = ({ weather, fetchWeather }) => { 
    return (
        <div className="weather-page-container">
            <h1 className="app-title">Audrey's Weather App</h1>
            
            <LocationForm fetchWeather={fetchWeather}/>
            <WeatherInfo weather={weather} />
        </div>
    );
};

export default Weather;