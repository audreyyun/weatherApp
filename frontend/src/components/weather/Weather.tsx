import React from 'react';
import styles from './Weather.module.scss';
import LocationForm from '../LocationForm/LocationForm';
import WeatherInfo from '../WeatherInfo/WeatherInfo';

const Weather = () => { 
    return (
        <div className={styles.weatherPageContainer}>
            <h1 className={styles.appTitle}>Audrey's Weather App</h1>
            
            <LocationForm/>
            {/* <WeatherInfo weather={weather} /> */}
        </div>
    );
};

export default Weather;