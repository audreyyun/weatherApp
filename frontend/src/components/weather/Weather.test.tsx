import '@testing-library/jest-dom/extend-expect';

import React from 'react'
import Weather from './Weather'
import { render } from '@testing-library/react'
import TestProvider from '../../__test__/TestProvider';


const WeatherTest = () => (
    <TestProvider>
        <Weather />
    </TestProvider>
);


describe('Weather', () => {
    it('should render without crashing', () => {
        const component = render(<WeatherTest />)
        expect(component.asFragment()).toMatchSnapshot()
    })
})
