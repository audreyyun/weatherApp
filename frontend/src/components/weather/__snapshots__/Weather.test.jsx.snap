// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="weatherPageContainer"
  >
    <h1
      class="appTitle"
    >
      Audrey's Weather App
    </h1>
    <div>
      <div
        class="locationForm"
      >
        <div
          class="locationInput"
        >
          <input
            class="zipcodeSearchInput input"
            placeholder=" Enter City Name"
          />
          <button
            class="locationInputBtn input"
          >
            See Weather
          </button>
        </div>
      </div>
    </div>
    <div
      class="no-weather-info"
    >
      Please enter a zip code and choose a country to see today's weather!
    </div>
  </div>
</DocumentFragment>
`;
